src/main/java/camp/woowak/lab/order/domain/StockRequester.java

```diff
@@ -1,34 +1,82 @@
 package camp.woowak.lab.order.domain;
 
+import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
 
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
 
 import camp.woowak.lab.cart.domain.vo.CartItem;
+import camp.woowak.lab.cart.exception.MenuNotFoundException;
+import camp.woowak.lab.infra.cache.MenuStockCacheService;
+import camp.woowak.lab.infra.cache.exception.CacheMissException;
 import camp.woowak.lab.menu.domain.Menu;
+import camp.woowak.lab.menu.exception.NotEnoughStockException;
 import camp.woowak.lab.menu.repository.MenuRepository;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 @Component
 public class StockRequester {
 	private final MenuRepository menuRepository;
+	private final MenuStockCacheService menuStockCacheService;
+	private final static int RETRY_COUNT = 10;
 
-	public StockRequester(MenuRepository menuRepository) {
+	public StockRequester(MenuRepository menuRepository, MenuStockCacheService menuStockCacheService) {
 		this.menuRepository = menuRepository;
+		this.menuStockCacheService = menuStockCacheService;
 	}
 
-	public void request(List<CartItem> cartItems) {
-		List<Long> cartItemMenuIds = extractMenuIds(cartItems);
-		List<Menu> allById = menuRepository.findAllByIdForUpdate(cartItemMenuIds);
-		for (Menu menu : allById) {
-			for (CartItem cartItem : cartItems) {
-				if (cartItem.getMenuId().equals(menu.getId())) {
-					menu.decrementStockCount(cartItem.getAmount());
+	/**
+	 *
+	 * @throw NotEnoughStockException 재고가 부족한 경우
+	 */
+	@Transactional
+	public List<CartItem> request(List<CartItem> cartItems) {
+		List<CartItem> stockDecreaseSuccessCartItems = new ArrayList<>();
+
+		// 메뉴 한개씩 돌면서 레디스 확인하고 디비 바꿔주고
+		// 레디스에 모두 돌고 모두 성공하면 한번에 디비에 업데이트
+		// redis cache에 메뉴 재고를 요청
+
+		for (CartItem cartItem : cartItems) {
+			Long menuId = cartItem.getMenuId();
+			for (int retryCount = 0; retryCount < RETRY_COUNT; retryCount++) {
+				try {
+					menuStockCacheService.addAtomicStock(menuId, -cartItem.getAmount());
+					stockDecreaseSuccessCartItems.add(cartItem);
+					break; // 재고 감소 성공 시 탈출
+				} catch (CacheMissException e) { // redis cache miss 면 findById 로 재고 확인 update
+					// (1) cache miss 시 RDB 에 접근해서 재고를 가져와야해요
+					// (2) 가져온 재고를 업데이트
+					Menu menu = menuRepository.findById(menuId)
+						.orElseThrow(() -> new MenuNotFoundException("메뉴가 존재하지 않습니다."));    // 준기: 10, 현수: 8 -> redis: 8
+					menuStockCacheService.updateStock(menuId, menu.getStockCount());    // lock
```

Comment:
```suggestion
				} catch (CacheMissException e) { // redis cache miss 면 findById 로 재고 확인 update
					// (1) cache miss 시 RDB 에 접근해서 재고를 가져와야해요
					// (2) 가져온 재고를 업데이트
					Menu menu = menuRepository.findById(menuId)
						.orElseThrow(() -> new MenuNotFoundException("메뉴가 존재하지 않습니다."));    // 준기: 10, 현수: 8 -> redis: 8
					menuStockCacheService.updateStock(menuId, menu.getStockCount());    // lock
					retryCount--;
```
`addAtomicStock()`을 시도할 때 redis에 값이 없으면 `CacheMissException`이 발생할 것 같네요.
그러면 1) menu의 재고를 DB에서 가져와서 2) 주문된 수량만큼 재고에서 차감하고, 3-1)재고가 부족하면 rollback 후 NotEnoughStockException` 3-2)재고가 충분하면 카트의 다음 메뉴의 재고 차감 시도 하는 플로우로 가야될 것 같습니다.

현재 CacheMissException이 발생한 경우 RDB의 데이터를 Redis에 업데이트는 하고 있는데 **재고를 차감하는 로직**이 빠진 것 같아요.